src/libraries/System.Security.Cryptography/ref/System.Security.Cryptography.cs

```diff
@@ -1801,21 +1801,31 @@ public abstract partial class MLKem : System.IDisposable
         protected MLKem(System.Security.Cryptography.MLKemAlgorithm algorithm) { }
         public System.Security.Cryptography.MLKemAlgorithm Algorithm { get { throw null; } }
         public static bool IsSupported { get { throw null; } }
+        public byte[] Decapsulate(byte[] ciphertext) { throw null; }
         public void Decapsulate(System.ReadOnlySpan<byte> ciphertext, System.Span<byte> sharedSecret) { }
+        public void Decapsulate(System.ReadOnlySpan<byte> ciphertext, System.Span<byte> sharedSecret, out int sharedSecretWritten) { throw null; }
         protected abstract void DecapsulateCore(System.ReadOnlySpan<byte> ciphertext, System.Span<byte> sharedSecret);
         public void Dispose() { }
         protected virtual void Dispose(bool disposing) { }
+        public byte[] Encapsulate(out byte[] sharedSecret) { throw null; }
```

Comment:
When doing some "write a demo program that uses ML-KEM", I felt like

```C#
public byte[] Encapsulate(Span<byte> sharedSecret);
```

was useful.  The CT into an array because I was about to send it over the network, and the secret to the stack so I could run it through my KDF then erase it.

It might not be all that useful in reality, though, since it's probably just a piece of a larger message, not just "here's my CT".

Reply:
"Oooh, maybe we should add an overload of encap/decap that don't write the shared secret at all, just pass it to a callback... since every valid use of the algorithm is 'KDF this'".

OK, maybe that's too far.  Since it would only compound the array vs span problem.

Reply:
No, because that means we will need a custom delegate for downlevel platforms since ref structs can't be type arguments, and new delegate types is like a 1000 hour discussion.

Reply:
> public byte[] Encapsulate(Span<byte> sharedSecret);

I thought about it and it feels weird to me. Since there is no `out`, then than means "the destination buffer must be exact in length". Which feels a weird mix of "power user" API and "allocate a thing for me".

So maybe you want one that has an `out int sharedSecretBytesWritten` makes it a little better.

Or maybe you think both are good?

Reply:
If you want to toss the bytesWritten into it, that's fine by me.  Or put both in and let API Review argue it down, that's fine, too.

But my suggested overload here is how I ended up deciding I liked `byte[] Encapsulate(out byte[])` over `(byte[] Ciphertext, byte[] SharedSecret) Encapsulate()`, because it looked more like this middle state.  (And because the two-ple is a footgun of backwardsness)

Reply:
Okay, I think this is fine.